test/integration/api_test.lua

```diff
@@ -1,12 +1,264 @@
+    g.cluster:stop()
+    fio.rmtree(g.cluster.datadir)
+end
+
+g.before_each(function()
+    for _, server in ipairs(g.cluster.servers) do
+        server.net_box:eval([[
+            local space = box.space.kv
+            if space ~= nil and not box.cfg.read_only then
+                space:truncate()
+            end
+        ]])
+    end
+end)
+
+g.after_each(function()
+
+end)
+
+-- CRUD helpers
+local function create(key, value)
+    return g.router:http_request('post', url..'/', {
+        json = {
+            key = key,
+            value = value,
+        },
+        http = { timeout = 1 },
+        raise = false,
+    })
+end
+
+local function read(key)
+    return g.router:http_request('get', url..'/'..key, { http = { timeout = 1 } , raise = false })
+end
+
+local function update(key, value)
+    local resp = g.router:http_request('put', url..'/'..key, {
+        json = { value = value },
+        http = {
+            timeout = 1
+        },
+        raise = false,
+    })
+    log.info(resp)
+    return resp
+end
+
+local function delete(key)
+    return g.router:http_request('delete', url..'/'..key, { http = { timeout = 1 }, raise = false })
+end
+
+
+-- Is request ok helpers
+local function ok(resp, expected_body)
+    t.assert_equals(resp.status, 200)
+    if expected_body ~= nil then
+        t.assert_equals(resp.json, expected_body)
+    end
+end
+
+local function create_ok(key, value)
+    ok(create(key, value), { result = 'record created' })
+end
+
+local function read_ok(key, expected_value)
+    ok(read(key), { value = expected_value })
+end
+
+local function update_ok(key, value)
+    ok(update(key, value), { result = 'record updated' })
+end
+
+local function delete_ok(key)
+    ok(delete(key), { result = 'record deleted' })
+end
+
+
+-- Is key already exists helpers
+local function already_exists(resp)
+    t.assert_equals(resp.status, 409)
+    t.assert_equals(resp.json, { error = 'key already exists' })
+end
+
+local function create_already_exists(key, value)
+    already_exists(create(key, value))
+end
+
+
+-- Is key not found helpers
+local function not_found(resp)
+    t.assert_equals(resp.status, 404)
+    t.assert_equals(resp.json, {error = 'key not found'})
+end
+
+local function read_not_found(key)
+    not_found(read(key))
+end
+
+local function update_not_found(key, value)
+    not_found(update(key, value))
+end
+
+local function delete_not_found(key)
+    not_found(delete(key))
+end
+
+
+g.test_read_not_found = function()
+    read_not_found('some_key')
+end
+
+g.test_update_not_found = function()
+    update_not_found('some_key', {'hello_world'})
+end
+
+g.test_delete_not_found = function()
+    not_found(delete('some_key'))
+end
+
+g.test_create_ok = function()
+    create_ok('awesome key', { some_json = { 'opt 1', 'opt 2', 'opt 3'} })
+end
+
+g.test_create_already_exists = function()
+    create_ok('some_key', {name = 'semen'})
+    create_already_exists('some_key', { array = {0, 1, 2, 3} })
```

Comment:
What do you think about smth like

```lua
local resp = create('some_key', {name = 'semen'})
assert_resp_ok(resp, <exp_body>)

local resp = create('some_key', {name = 'semen'})
assert_err_already_exists(resp)
```

It looks that `<*_not_found >` functions is too much, it's enough to separate functions for http requests, and functions to check response.